src/encoder.rs

```diff
@@ -179,6 +196,35 @@ impl<W: Write> Writer<W> {
             write_chunk(&mut self.w, chunk::tRNS, t)?;
         }
 
+        if let Some(g) = &self.info.source_gamma {
+            let scale_factor = 100000.;
+            let g = (g * scale_factor).floor() as u32;
+            write_chunk(&mut self.w, chunk::gAMA, &g.to_be_bytes())?;
+        }
+
+        if let Some(c) = &self.info.primary_chromaticities {
+            let scale_factor = 100000.;
+            let white_x = ((c.white_point.0 * scale_factor).round() as u32).to_be_bytes();
+            let white_y = ((c.white_point.1 * scale_factor).round() as u32).to_be_bytes();
+            let red_x = ((c.red.0 * scale_factor).round() as u32).to_be_bytes();
+            let red_y = ((c.red.1 * scale_factor).round() as u32).to_be_bytes();
+            let green_x = ((c.green.0 * scale_factor).round() as u32).to_be_bytes();
+            let green_y = ((c.green.1 * scale_factor).round() as u32).to_be_bytes();
+            let blue_x = ((c.blue.0 * scale_factor).round() as u32).to_be_bytes();
+            let blue_y = ((c.blue.1 * scale_factor).round() as u32).to_be_bytes();
+            let enc = [
+                white_x[0], white_x[1], white_x[2], white_x[3],
+                white_y[0], white_y[1], white_y[2], white_y[3],
+                  red_x[0],   red_x[1],   red_x[2],   red_x[3],
+                  red_y[0],   red_y[1],   red_y[2],   red_y[3],
+                green_x[0], green_x[1], green_x[2], green_x[3],
+                green_y[0], green_y[1], green_y[2], green_y[3],
+                 blue_x[0],  blue_x[1],  blue_x[2],  blue_x[3],
+                 blue_y[0],  blue_y[1],  blue_y[2],  blue_y[3],
+            ];
```

Comment:
With respect to `cargo fmt` complaining about the formatting here, use `#[rustfmt::skip]`. The tool is, sadly, rather tone deaf to good matrix formatting.

Reply:
Yeah, its the same in all major editors right. I'll add the #[rustfmt::skip] annotation.

Reply:
I moved the specific code to a separate method and put the #[rustfmt::skip] annotation on that method only to avoid having to skip formatting on the entire init() method. Cargo fmt now causes no re-formatting anymore.